examples/basic_setup/src/main.rs

```diff
@@ -0,0 +1,45 @@
+use fantoccini::error::CmdError;
+use fantoccini::{Client, Locator};
+use futures::future::Future;
+use tokio;
+
+fn main() {
+    // expects WebDriver instance to be listening at port 4444
+    let client = Client::new("http://localhost:4444");
+    let get_started_button =
+        r#"//a[@class="button button-download ph4 mt0 w-100" and @href="/learn/get-started"]"#;
+    let try_without_installing_button =
+        r#"//a[@class="button button-secondary" and @href="https://play.rust-lang.org/"]"#;
+    let play_rust_lang_run_button = r#"//div[@class="segmented-button"]/button[1]"#;
+
+    let rust_lang = client
+        .map_err(|error| unimplemented!("failed to connect to WebDriver: {:?}", error))
+        .and_then(|client| client.goto("https://www.rust-lang.org/"))
+        .and_then(|client| client_wait(client, 3000))
```

Comment:
I don't think you should need all these waits :)

Reply:
Oh yeah I know it isn't necessary, but if I don't put it there to the example flys by. You're talking about the `client_wait` right?

Reply:
Oh, they're there to let the user observe the browser changing? Might be worth commenting that here so that people don't start using this pattern everywhere in their own code!

Reply:
How would I go about doing that. Should I warn them not to use it or suggest to wait till a native solution comes along?

Reply:
I would just put a comment in the first time you use it:
```rust
        .and_then(|client| {
            // we add an artificial delay here so that the example is easier to watch
            // you do not need these delays in your own code
            client_wait(client, 3000))
        })
```